api/src/main/java/org/apache/iceberg/io/RangeReadable.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.io;
+
+/**
+ * {@code RangeReadable} is an interface that allows for implementations
+ * of {@link InputFile} streams to perform positional, range-based reads, which
+ * are more efficient than unbounded reads in many cloud provider object stores.
+ *
+ * Thread safety is not a requirement of the interface and is left to the
+ * implementation.
+ *
+ * If the implementation is also a {@link SeekableInputStream}, the position
+ * of the stream is not required to be updated based on the positional reads
+ * performed by this interface.  Usage of {@link SeekableInputStream} should
+ * always seek to the appropriate position for {@link java.io.InputStream}
+ * based reads.
+ *
+ */
+public interface RangeReadable extends AutoCloseable {
```

Comment:
We usually extend `Closeable` rather than `AutoCloseable`. That's because most places check for `Closeable` and we would rather always pass the check. Plus, `InputStream` implements `Closeable`.